domain/src/main/java/org/oppia/android/domain/feedbackreporting/FeedbackReportManagementController.kt

```diff
@@ -0,0 +1,346 @@
+import java.io.IOException
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private const val FEEDBACK_REPORT_MANAGEMENT_CONTROLLER_TAG =
+  "Feedback Report Management Controller"
+private const val FEEDBACK_REPORTS_DATABASE_NAME = "feedback_reports_database"
+private const val PLATFORM_ANDROID = "android"
+
+/** Controller for uploading feedback reports to remote storage or saving them on disk. */
+@Singleton
+class FeedbackReportManagementController @Inject constructor(
+  cacheStoreFactory: PersistentCacheStore.Factory,
+  private val feedbackReportingService: FeedbackReportingService,
+  private val consoleLogger: ConsoleLogger,
+  private val exceptionsController: ExceptionsController,
+  private val analyticsController: AnalyticsController,
+  private val networkConnectionUtil: NetworkConnectionUtil,
+  @ReportSchemaVersion private val reportSchemaVersion: Int
+) {
+  private val feedbackReportDataStore = cacheStoreFactory.create(
+    FEEDBACK_REPORTS_DATABASE_NAME, FeedbackReportingDatabase.getDefaultInstance()
+  )
+
+  private val oppiaEventLogsLiveData by lazy {
+    analyticsController.getEventLogStore().toLiveData()
+  }
+
+  private val eventsLogListLiveData = Transformations.map(oppiaEventLogsLiveData) {
+    it.getOrDefault(OppiaEventLogs.getDefaultInstance())
+  }
+
+  private val eventsLogList = Transformations.map(eventsLogListLiveData) { oppiaEventLogs ->
+    oppiaEventLogs.eventLogList.map { it.toString() }
+  }.value
+
+  /**
+   * Submits a [FeedbackReport] to remote storage to be processed in the admin dashboard, or saves it
+   * to local storage if there is no internet connection available to send reports.
+   *
+   * @param feedbackReportViewModel for the view model representing the feedback provided by the user
+   */
+  fun submitFeedbackReport(feedbackReport: FeedbackReport) {
+    if (networkConnectionUtil.getCurrentConnectionStatus() == NONE) {
+      storeFeedbackReport(feedbackReport)
+    } else {
+      uploadFeedbackReport(feedbackReport)
+    }
+  }
+
+  /**
+   * Gets any feedback reports saved locally on the device. This is called by a
+   * [FeedbackReportUploadWorker] when it detects a network reconnection to send all local reports
+   * to remote storage.
+   *
+   * @return a list of feedback reports to upload to remote storage
+   */
+  suspend fun getCachedReportsList(): MutableList<FeedbackReport> {
+    return feedbackReportDataStore.readDataAsync().await().reportsList
+  }
+
+  /**
+   * Gets the data store of all locally-saved feedback reports.
+   *
+   * @return the data provider for the storage database
+   */
+  fun getFeedbackReportStore(): DataProvider<FeedbackReportingDatabase> {
+    return feedbackReportDataStore
+  }
+
+  /**
+   * Sends an individual [FeedbackReport] to remote storage.
+   *
+   * @param report a [FeedbackReport] to upload
+   */
+  fun uploadFeedbackReport(report: FeedbackReport) {
```

Comment:
Should this be private? It seems like we'd always want reports to go through submitFeedbackReport().

If not (particularly because we want to log things through the worker), then we should probably have a few targeted tests that verify this always forces a send regardless of internet connectivity (or put another way: the caller is responsible for calling this in the correct circumstances).